src/infra/src/table/migration/m20250115_000001_create_timed_annotations_table.rs

```diff
@@ -0,0 +1,111 @@
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+const ANNOTATIONS_DASHBOARD_ID_IDX: &str = "timed_annotations_dashboard_id_idx";
+const TIMED_ANNOTATIONS_DASHBOARDS_FK: &str = "fk_timed_annotation_panels_dashboard_id";
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager.create_table(create_table_stmt()).await?;
+        manager
+            .create_index(create_index_dashboard_id_stmt())
+            .await?;
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .drop_index(Index::drop().name(ANNOTATIONS_DASHBOARD_ID_IDX).to_owned())
+            .await?;
+        manager
+            .drop_table(Table::drop().table(TimedAnnotations::Table).to_owned())
+            .await?;
+        Ok(())
+    }
+}
+
+/// Statement to create table.
+fn create_table_stmt() -> TableCreateStatement {
+    Table::create()
+        .table(TimedAnnotations::Table)
+        .if_not_exists()
+        // The ID is 27-character human readable KSUID.
+        .col(
+            ColumnDef::new(TimedAnnotations::Id)
+                .char_len(27)
+                .not_null()
+                .primary_key(),
+        )
+        .col(
+            ColumnDef::new(TimedAnnotations::DashboardId)
+                .char_len(27)
+                .not_null()
+        )
+        .col(
+            ColumnDef::new(TimedAnnotations::StartTime)
+                .big_integer()
+                .not_null(),
+        )
+        .col(
+            ColumnDef::new(TimedAnnotations::EndTime)
+                .big_integer()
+                .null(),
+        )
+        .col(
+            ColumnDef::new(TimedAnnotations::Title)
+                .string_len(256)
+                .not_null(),
+        )
+        .col(ColumnDef::new(TimedAnnotations::Text).text().null())
+        .col(
+            ColumnDef::new(TimedAnnotations::Tags)
+                .json()
+                .not_null()
+                .default(serde_json::json!([])),
```

Comment:
Will this work on all databases? If so, then great! That's very convenient.

I'd be curious to see what kind of SQL this gets translated into for different databases. I'd be a little worried that at least one of the databases we support might not support this in the SQL generation, but we can certainly test that to confirm. If they all support setting a default JSON value on a `json` column then this is very nice. 👍

Reply:
You're right, it doesn't work for `mysql`. However, it works for `pg` and `sqlite`